manim/utils/hashing.py

```diff
@@ -30,56 +32,122 @@ def default(self, obj):
             Python object that JSON encoder will recognize
 
         """
-        if inspect.isfunction(obj) and not isinstance(obj, ModuleType):
+        if not (isinstance(obj, ModuleType)) and isinstance(
+            obj, (MethodType, FunctionType)
+        ):
             cvars = inspect.getclosurevars(obj)
             cvardict = {**copy.copy(cvars.globals), **copy.copy(cvars.nonlocals)}
             for i in list(cvardict):
                 # NOTE : All module types objects are removed, because otherwise it throws ValueError: Circular reference detected if not. TODO
                 if isinstance(cvardict[i], ModuleType):
                     del cvardict[i]
-            return {"code": inspect.getsource(obj), "nonlocals": cvardict}
+            return self._check_iterable(
+                {"code": inspect.getsource(obj), "nonlocals": cvardict}
+            )
         elif isinstance(obj, np.ndarray):
-            return list(obj)
+            if obj.size > 1000:
+                return f"Np array too big {obj.size}"
+            # We return the repr and not a list to avoid the JsonEncoder to iterate over it.
+            return repr(obj)
         elif hasattr(obj, "__dict__"):
             temp = getattr(obj, "__dict__")
-            return self._encode_dict(temp)
+            # MappingProxy is not supported by the Json Encoder
+            if isinstance(temp, MappingProxyType):
+                return "Unsupported type for serializing ->" + str(type(temp))
```

Comment:
You can convert a `MappingProxy` to a dictionary and use that, IIRC.

Reply:
I will look into that. Good suggestion !